### PR TITLE
Localstorage

### PR DESCRIPTION
**Pull request by ctokoli**
- Store data in local storage
- Store data in a single javascript object
- Show data in the input when the page reloads

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -98,3 +101,41 @@ function getData(featuredImage, live, source) {
...
+document.querySelector('.form-view').addEventListener('submit', storeData);
+
+function loadData() {
+  const userData = JSON.parse(localStorage.getItem('formdata') || '[]');
+  if (userData !== 'undefined' || userData !== null) {
+    nameValue.value = userData.name;
+    emailValue.value = userData.email;
+    msgValue.value = userData.message;
+  }
+}
```
Comment (christianbiring1) [inline script.js]: - [ ] As you may notice how the prefilled values of your inputs are `undefined` when the page load. Please make the right data are saved or retrieved from your `localStorage.` as you may see the string value `'[ ]'` instead if there is nothing from the storage which there can be no way it has some values like `userData.name`,` ...`
YOu may consider deleting that `'[ ]'` and also say something like nameValue = userData.name || `" "`
image

File: script.js [hunk 2]
```diff
@@ -98,3 +101,41 @@ function getData(featuredImage, live, source) {
...
+function storeData() {
+  const formData = {
+    name: nameValue.value,
+    email: emailValue.value,
+    message: msgValue.value,
+  };
+  localStorage.setItem('formdata', JSON.stringify(formData));
+}
+
+document.querySelector('.form-view').addEventListener('submit', storeData);
```
Comment (christianbiring1) [inline script.js:130]: - [ ] Also make sure the data in the storage is in sync as the user is typing in the input fields. means the data are not supposed to be saved only when the user submits the form but when the user starts typing in the field.
Hint; You may consider using an `Input` event.

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @ctokoli , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle: 


## Highlights
- [ ] Great work on saving the user inputs correctly in the `localStorage` :+1: 
- [ ] All linters checks are green :heavy_check_mark: 
- [ ] Descriptive PR `title` and `summary`

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @ctokoli , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done implementing the requested changes. ✔
- You successfully completed all the requirements.  🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**